pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/Common.java

```diff
@@ -501,6 +501,168 @@ private static String getRepeatedEqualsGeneration(String generatedCodeSoFar, Fie
 		return generatedCodeSoFar;
 	}
 
+
+	/**
+	 * Generate the compareTo method content for the provided fields
+	 * @param fields The fields of this object.
+	 * @param generatedCodeSoFar the generated code so far (non-empty in case of nested objects)
+	 * @return The generated code for compareTo method body
+	 */
+	public static String getFieldsCompareToStatements(final List<Field> fields, String generatedCodeSoFar) {
+		for(Field f : fields) {
+			if (f.parent() != null) {
+				final OneOfField oneOfField = f.parent();
+				generatedCodeSoFar += getFieldsCompareToStatements(oneOfField.fields(), generatedCodeSoFar);
+			}
+
+			if (f.optionalValueType()) {
+				generatedCodeSoFar += getPrimitiveWrapperCompareToGeneration(f);
+			}
+			else if (f.repeated()) {
+				// intentionally skipping repeated fields
+			} else {
+				f.nameCamelFirstLower();
+				if (f.type() == Field.FieldType.FIXED32 ||
+						f.type() == Field.FieldType.INT32 ||
+						f.type() == Field.FieldType.SFIXED32 ||
+						f.type() == Field.FieldType.SINT32 ||
+						f.type() == Field.FieldType.UINT32) {
+					generatedCodeSoFar +=
+							"""
+                            result = Integer.compare($fieldName, thatObj.$fieldName);
```

Comment:
I wonder how `Integer.compare()` handles unsigned 32-bit ints...

Reply:
It doesn't (at least not correctly).
Here's the full code of Integer.compare (it's quite simple):
`return (x < y) ? -1 : ((x == y) ? 0 : 1);`
It compares integers as signed.
If the values are unsigned, then this would be:
`result = Integer.compareUnsigned($fieldName, thatObj.$fieldName);`

The long.compare code below should be similar with unsigned fields separated to use `Long.compareUnsigned()`.

Reply:
Fixed